skills_ml/algorithms/corpus_creators/basic.py

```diff
@@ -91,12 +90,68 @@ class GensimCorpusCreator(CorpusCreator):
     ]
     join_spaces = ' '.join
 
+    def __init__(self, generator=None):
+        super().__init__()
+        self.lookup = {}
+        self.generator = generator
+        self.k = 0
+
     def _transform(self, document):
         return self.join_spaces([
             self.nlp.clean_str(document[field])
             for field in self.document_schema_fields
         ])
 
+    def __iter__(self):
+        for line in self.generator:
+            document = json.loads(line)
+            # Only train on job posting that has onet_soc_code
+            if document['onet_soc_code']:
```

Comment:
Can you explain what's going on here with the tagging? I see that we're using self.k as an accumulating number, and we are saving the given ONET code in the object with the number. But we are also sending the accumulating number in an array to TaggedDocument. What is its purpose there?